judo-runtime-core-jsl-itest/models/PagingModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/PagingTest.java

```diff
@@ -0,0 +1,422 @@
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.paging.paging.list.ListIdentifier;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.paging.paging.logentry.*;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.guice.PagingDaoModules;
+import hu.blackbelt.judo.requirement.report.annotation.Requirement;
+import hu.blackbelt.judo.requirement.report.annotation.TestCase;
+import hu.blackbelt.judo.runtime.core.jsl.AbstractJslTest;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+
+import java.time.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Slf4j
+public class PagingTest extends AbstractJslTest {
+    @Inject
+    ItemDao itemDao;
+
+    @Inject
+    ListDao listDao;
+
+    @Inject
+    LogEntryDao logEntryDao;
+
+
+    @Override
+    public Module getModelDaoModule() {
+        return new PagingDaoModules();
+    }
+
+    @Override
+    public String getModelName() {
+        return "Paging";
+    }
+
+    /**
+     * Testing the limit and offset in bigger data and combine other query customizer methods (filter, order).
+     *
+     * @prerequisites The model runtime is empty. It means that the database of the application has to be empty before this test starts running.
+     *
+     * @type Behaviour
+     *
+     * @others Implement this test case in the *judo-runtime-core-jsl-itest* module.
+     *
+     * @jslModel Paging.jsl
+     *
+     */
+    @Test
+    @TestCase("Windowing")
+    @Requirement(reqs = {
+            "REQ-MDL-001",
+            "REQ-MDL-002",
+            "REQ-MDL-003",
+            "REQ-TYPE-001",
+            "REQ-TYPE-004",
+            "REQ-TYPE-005",
+            "REQ-ENT-001",
+            "REQ-ENT-002",
+            "REQ-ENT-008",
+            "REQ-EXPR-001",
+            "REQ-EXPR-003",
+            "REQ-EXPR-006",
+            "REQ-EXPR-021"
+    })
+    public void testWindowing() {
+
+        Map<Character, ListIdentifier> listIds = new TreeMap<>();
+        for (char ch = 'A'; ch <= 'Z'; ch++) {
+            final java.util.List<Item> items = new ArrayList<>();
+            for (int i = 0; i < 'Z' - ch + 1; i++) {
+                final String topic;
+                if (ch < 'E' && i < 3) {
+                    topic = "Topic" + (char) (ch + i);
+                } else {
+                    topic = null;
+                }
+                items.add(Item.builder().withNumber(i + 1).withTopic(topic).build());
+            }
+            List list = listDao.create(List.builder().withName("List_" + ch).withItems(items).build());
+
+            listIds.put(ch, list.identifier());
+        }
```

Comment:
![wtf-gif](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/assets/54545024/3b5dc3f3-cbf3-4343-b58e-86bfdf044570)
:D

Reply:
This came from the esm-itest and made be Robi :D